src/allmydata/test/test_node.py

```diff
@@ -513,103 +519,106 @@ def test_set_config_rejects_invalid_config(self):
 
 class TestMissingPorts(unittest.TestCase):
     """
-    Test certain error-cases for ports setup
+    Test certain ``_tub_portlocation`` error cases for ports setup.
     """
-
     def setUp(self):
         self.basedir = self.mktemp()
         create_node_dir(self.basedir, "testing")
 
-    def test_parsing_tcp(self):
+    def _get_addr(self):
+        return ["LOCAL"]
```

Comment:
I know that this documented in the public code, but with the new name, I think that this should have a docstring of

```
A function like ``iputil.get_local_addresses_sync``.
```

My suggestion is to move it ouside of the class...since `self` is not used and rename it to

iputil_get_local_addresses_sync_local and iputil_allocate_tcp_port_999

It should be easier to read the tests as you know that it will return 999 or local.

If more values are required, you can go wild and have it as a function generator `iputil_get_local_addresses_sync(['LOCAL'])`